### PR TITLE
Nerfs Piggybacking

### PR DESCRIPTION
**Pull request by Obliii**
As it stands currently, being able climb on someone's back in 1.5s while in a combat scenario seems to be something that I constantly see happen. 

Let's increase it to say 4 seconds so that I'm able to choke someone or slam them somewhere without having to worry about them climbing onto MY FUCKING BACK to then unbuckle and run off!

## About The Pull Request

Changes the amount of time needed to piggyback onto someone from 1.5s > 4s. 
(Since quickly carrying takes about that much time.)

## How This Contributes To The Skyrat Roleplay Experience

Being able to climb onto someone's back while being aggro grabbed in a combat scenario has always seemed kind of silly. Especially since you can do that and immediately unbuckle to run off into the horizon! 

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

https://github.com/Skyrat-SS13/Skyrat-tg/assets/46305948/03888fab-7dd6-4140-b0bf-9af41474ecdd

</details>

## Changelog

:cl:
balance: Changed time for Piggybacking from [ 1.5s > 4s ].
/:cl:

### CODE REVIEW

**Comment by Nerev4r** [comment]
for posterity cloning this comment here
you can hit rest when someone succeeds in climbing on your back and it basically stuns them
not only that but before they get on, you can just. move. or drag them and start kicking them, in which kicking does increased stamina and brute. 
and the main thing is
climbing as a grab counter is necessary because otherwise grabbing is just a game of 'keep grabbing until you achieve success' because resisting out of a grab costs stamina


edit: it's also now easier than ever to get grabs post-brawlening, so it's double needed. not to mention you cannot begin climbing someone if you have too little stamina, and pummeling/kicking nukes said stamina. most i'd be comfortable with is two seconds.

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -966,10 +966,12 @@
 		to_chat(target, span_warning("You can't piggyback ride [src] right now!"))
 		return
 
+	//SKYRAT EDIT START
 	visible_message(span_notice("[target] starts to climb onto [src]..."))
-	if(!do_after(target, 1.5 SECONDS, target = src) || !can_piggyback(target))
+	if(!do_after(target, 4.0 SECONDS, target = src) || !can_piggyback(target))
 		visible_message(span_warning("[target] fails to climb onto [src]!"))
 		return
+	//SKYRAT EDIT END
```
Comment (SomeRandomOwl) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
	if(!do_after(target, 4.0 SECONDS, target = src) || !can_piggyback(target)) //SKYRRAT EDIT - ORIGINAL: 1.5 SECONDS
		visible_message(span_warning("[target] fails to climb onto [src]!"))
		return
```

This is the proper format for a non-modular edit like this

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -966,10 +966,12 @@
 		to_chat(target, span_warning("You can't piggyback ride [src] right now!"))
 		return
 
+	//SKYRAT EDIT START
 	visible_message(span_notice("[target] starts to climb onto [src]..."))
```
Comment (SomeRandomOwl) [inline code/modules/mob/living/carbon/human/human.dm:970]:
```suggestion
	visible_message(span_notice("[target] starts to climb onto [src]..."))
```

**Review by SomeRandomOwl — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (SomeRandomOwl) [reply code/modules/mob/living/carbon/human/human.dm:970]: Don't forget this one too, as the edit start isn't necessary for something that is changing the value of a single line

**Comment by chemistrymain2** [comment]
This is unnecessary and also an intended counter to aggro-grabbing, as resisting out of a grab costs stamina.

This is part of maint-fu and is intentional behavior, low cooldown and all.

**Comment by kittysmooch** [comment]
echoing previous sentiments, this is more of an unbalance pr. you should take more time in the kitchen on this one.

**Comment by Iamgoofball** [comment]
>This is part of maint-fu and is intentional behavior, low cooldown and all.

Do you have a source that this is intended maint-fu behavior?

**Comment by Namwty** [comment]
Look chaps, I think this is a good change, I'm sorry, but this "mechanic" ( which I highly doubt is intended ) gives me/others more reason to just off people instead of trying to aggro grab handcuffs, after stunlocking be it from stunstick or punching them.

Reality is, cause of how layer priority works, wings and so on... this is just unhealthy and frustrating. Imagine being above 1 scale size and having some 0.84 scale caracter climb on you and be literally hidden by like wings/tail/layering and shit, making you hit yourself. At that point people rather just not risk aggro grabbing and just killing them instead.

Now I never had this issue ever happening to me, and I'm more of "melee combat" than "ranged weapon", so frankly I couldn't give a shit tbf cause either way, this won't effect me. But I have witnessed it and I had a person complain about it, and it went something along "It was cheesey, and I should have just, killed em instead", but in more "happy" tone if ya get what I mean, so it is something that is frustrating to deal with, for some players.

Piggyback always seemed like "RP" feature, more than anything. If it was combat, then surely we could just have stuff like jousting mechanic be enable to everyone who does it ( not just borgs or whatever else ).

Maint-fu behavior or not, it doesn't justify it being dogshit, cause of it not giving you a real warning nor being fair to anyone who uses wings/tails and size scaling. Want it to be main-fu, feel free to add fancy CQC-ish lines/sounds of someone trying to climb and crap, make it at least more plausible than "haha I'm a friend who just likes to climb on someone's back"

EDIT: I won't talk about this "counter" stuff that people are talking, I'm sorry but you aren't going to play 5d chess if said person you are trying to aggro grab has a ally, it is just unrealistic, at that point you will feel like ass for not just killing them in first place. And if piggyback is RP mechanic, and we allow it to be used in combat, why not allow using pool tiles in combat, same deal both are RP oriented and "makes sense". Why not let AI's just build pool tables and stuff like good old days?

Like do we really need to have oversized characters to abuse this crap, to show how dumb something like this is? Ideally we just have this patched out. If this is a feature, then expect at one point oversized characters to just do 25 brute + wounds everytime they get grabbed. ( cause oversized characters piggyback = vendimachine-ish crushing you ). Just cause someone doesn't abuse the mechanic, doesn't mean it's ok to have it in first place. ( Altough same can be said about pool tiles, so who knows, either way, I don't care. But if this ever becomes a true meta in combat, it will be ass ).

EDIT2: And what is this about stunlocks and little stamina, for all I care, as long as you aren't 162/stamcrit stamina dmg or whatever it is, you can climb on the person, same as you can instantly hit resist to avoid being "stunlocked", cause you just go off them by yourself and being "tossed off".

**Comment by FieryMorty** [comment]
> climbing as a grab counter is necessary

I highly disagree and I do not think that this RP mechanic should be used in PVP.

That is unironically power-gaming, to use what was intended for RP, in PVP.

Especially when a small character can fail an attempt to get out normally, climb simultaneously to avoid having to wait for another chance to escape by escaping with WASD, climb onto an oversized person, have broken layering on top of him and hide under him, and then keep attacking him.

Silly and TG would never accept something like that and we shouldn't either.

**Comment by YakumoChen** [comment]
> for posterity cloning this comment here you can hit rest when someone succeeds in climbing on your back and it basically stuns them not only that but before they get on, you can just. move. or drag them and start kicking them, in which kicking does increased stamina and brute. and the main thing is climbing as a grab counter is necessary because otherwise grabbing is just a game of 'keep grabbing until you achieve success' because resisting out of a grab costs stamina
> 
> edit: it's also now easier than ever to get grabs post-brawlening, so it's double needed. not to mention you cannot begin climbing someone if you have too little stamina, and pummeling/kicking nukes said stamina. most i'd be comfortable with is two seconds.

It sounds like your justification for this obscure and probably unintended exploit of grab mechanics is another equally obscure grab mechanic that is unhealthy to the game. You're expected to lie down in combat to prevent someone from abusing a weird rp grab mechanic? Or "just kill them lmao"????

**Comment by Nerev4r** [comment]
> > for posterity cloning this comment here you can hit rest when someone succeeds in climbing on your back and it basically stuns them not only that but before they get on, you can just. move. or drag them and start kicking them, in which kicking does increased stamina and brute. and the main thing is climbing as a grab counter is necessary because otherwise grabbing is just a game of 'keep grabbing until you achieve success' because resisting out of a grab costs stamina
> > edit: it's also now easier than ever to get grabs post-brawlening, so it's double needed. not to mention you cannot begin climbing someone if you have too little stamina, and pummeling/kicking nukes said stamina. most i'd be comfortable with is two seconds.
> 
> It sounds like your justification for this obscure and probably unintended exploit of grab mechanics is another equally obscure grab mechanic that is unhealthy to the game. You're expected to lie down in combat to prevent someone from abusing a weird rp grab mechanic? Or "just kill them lmao"????

Or just pull them and move them.
Until resisting doesn't drain your stamina, this is the most equitable counter to grabs post-brawlening; need I remind that 

add: Someone shoved off-balance makes them vulnerable to more guaranteed unarmed strikes, knockdowns from a successful punch, and more difficult to escape grabs.
add: Grabbing someone (as well as kicking them while they're on the floor) makes them more vulnerable to taking unarmed attack damage, even if they have armor.

The most I'm comfortable with is two seconds. If people don't like that it's physically climbing onto someone, then I'd want a new move added that does a leg sweep or whatever.

**Comment by chemistrymain2** [comment]
We want to keep unarmed combat as flexible and unique as possible, especially with grabs taking stamina to resist out of.

This behavior is fine. Moreover, it is not a balance concern.

If you want to look at something that's objectively overpowered, go have fun trying to balance traitor revolvers. This, in particular, is a PR about balance where we don't really concern ourselves with balance unless it's extraordinarily excessive.

**Comment by YakumoChen** [comment]
> Or just pull them and move them.
Until resisting doesn't drain your stamina, this is the most equitable counter to grabs post-brawlening; need I remind that

I don't see the problem with resisting draining stamina, the alternative is punching people to crit and subduing them that way. Considering grabs and shoves are non-lethal, the alternative you want to make untenable is the purely pacifist option.

> If you want to look at something that's objectively overpowered, go have fun trying to balance traitor revolvers.

Did you actually mean to say: Guncargo?

**Comment by Nerev4r** [comment]
> > Or just pull them and move them.
> > Until resisting doesn't drain your stamina, this is the most equitable counter to grabs post-brawlening; need I remind that
> 
> I don't see the problem with resisting draining stamina, the alternative is punching people to crit and subduing them that way. Considering grabs and shoves are non-lethal, the alternative you want to make untenable is the purely pacifist option.
> 
> > If you want to look at something that's objectively overpowered, go have fun trying to balance traitor revolvers.
> 
> Did you actually mean to say: Guncargo?

You just have to put them even remotely near stamcrit and they can't climb you. Hell, you can probably get off a cuff before then.

**Comment by YakumoChen** [comment]
I'm not a fan of the word "probably" in this case if they can "maybe" escape a situation through a shitty obscure mechanic if you already have them dead to rights

The ardent defenders of this are also overlooking the clear abuse of size and layering going on here

**Comment by YakumoChen** [comment]
This also needs more visibility, piggybacking should shiw bubble text to the one being climbed if it doesn't already (pretty sure it does not)

**Comment by YakumoChen** [comment]
Unsaid before but piggybacking would make neck grabs impossible without enough stamina damage if the other player knows this move, which is kinda odd since neck grabs are already very hard to do

**Comment by desertmote** [comment]
Non-modular edit.
Did you look at the code below your edit? A piggyback attempt will fail when a user has `TRAIT_INCAPACITATED`. This trait is given to players in `stam_crit`.
If you want a player to be stunlocked, you will need to move beyond aggressive grabbing. This PR seems to me like you're trying to achieve a stunlock with an aggressive grab, but that's not their use-case.

Lastly TG has recently overhauled grabbing and tackle'ing, security now begins with tackle gloves who are capable of rolling an aggressive grab. Another aggressive grab buff is unnecessary, putting a victim at exclusively the mercy of random% rolling a resist attempt isn't better game-design than having a "silly" telegraphed `do_after` that can already easily be canceled.

**Comment by tf-4** [comment]
Just shove them when they try to climb, it interrupts their timer and if you do it into a wall it doesn't break the grab

**Comment by projectkepler-ru** [comment]
huh

**Comment by kittysmooch** [comment]
> Look chaps, I think this is a good change, I'm sorry, but this "mechanic" ( which I highly doubt is intended ) gives me/others more reason to just off people instead of trying to aggro grab handcuffs, after stunlocking be it from stunstick or punching them.

This is a dry snitch.

**Comment by kittysmooch** [comment]
> > climbing as a grab counter is necessary
> 
> I highly disagree and I do not think that this RP mechanic should be used in PVP.
> 
> That is unironically power-gaming, to use what was intended for RP, in PVP.
> 
> Especially when a small character can fail an attempt to get out normally, climb simultaneously to avoid having to wait for another chance to escape by escaping with WASD, climb onto an oversized person, have broken layering on top of him and hide under him, and then keep attacking him.
> 
> Silly and TG would never accept something like that and we shouldn't either.

referencing the original pr, which came from TG, it is intended to work this way

**Comment by kittysmooch** [comment]
>escape a situation through a shitty obscure mechanic if you already have them dead to rights

having someone "dead to rights" with two ctrl+clicks is pretty goofy from a balance perspective, why do you want this?

**Comment by Obliii** [comment]
Seeing this all has been an experience, and I understand somewhat of what everyone is saying. I wonder if there's a way to make everyone happy in this scenario.
Reducing the time to 2 or 2.5 seconds so that it would allow a very small window for the neckgrab if you can't climb or resist out.

**Comment by desertmote** [comment]
> Seeing this all has been an experience, and I understand somewhat of what everyone is saying. I wonder if there's a way to make everyone happy in this scenario.
> 
> Reducing the time to 2 or 2.5 seconds so that it would allow a very small window for the neckgrab if you can't climb or resist out. 

Leave it as is and apply stamina crit before trying to handcuff.

**Comment by gamermomentum** [comment]
I'm baffled this is even an argument. 

This mechanic is stupid when used for combat like this, because it invalidates the aggressive grab moves(Which I think is just suplex these days) and is mostly just for punching down on new people who don't know the mechanic. Batonning a guy for like ~20 straight seconds so you can throw cuffs on him, because he insists on using piggyback to try to game away, sucks. 
Trying to stamcrit a guy without a stun baton/sec shit so you can cuff them without critting them *also* sucks. 
And critting a guy to cuff him is kinda lame for roleplay, too. 

If you want a compromise with those people who think it should say, just add stamdamage to getting shoved off of somebody's back(When somebody right-clicks the person piggybacking them) so you can counter the "counter".

**Comment by Zergspower** [comment]
Frankly if this is an issue it can be taken upstream, i dont see this being a prevalent problem to make a modular edit

**Comment by Gandalf2k15** [comment]
Hi :) As this is an edit to TG code balance, taking it upstream is the best thing to do. As we are currently operating a skeleton crew, we do not have the facilities to be in-house testing and moderating balance changes to TG. Thanks for the PR though!

**Pull request closed without merging**